File: generated-sources/dart/mojang-sessions/lib/model/player_profile_texture_property_value.dart
part of openapi.api;

class PlayerProfileTexturePropertyValue {
  /* UNIX timestamp in milliseconds */
  int timestamp = null;
  /* The player UUID without hyphens */
  String profileId = null;
  /* The player's name */
  String profileName = null;
  /* Only present if unsigned was set to false in the request */
  bool signatureRequired = false;
  
  PlayerTexture textures = null;
  PlayerProfileTexturePropertyValue();

  @override
  String toString() {
    return 'PlayerProfileTexturePropertyValue[timestamp=$timestamp, profileId=$profileId, profileName=$profileName, signatureRequired=$signatureRequired, textures=$textures, ]';
  }

  PlayerProfileTexturePropertyValue.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    timestamp = json['timestamp'];
    profileId = json['profileId'];
    profileName = json['profileName'];
    signatureRequired = json['signatureRequired'];
    textures = new PlayerTexture.fromJson(json['textures']);
  }

  Map<String, dynamic> toJson() {
    return {
      'timestamp': timestamp,
      'profileId': profileId,
      'profileName': profileName,
      'signatureRequired': signatureRequired,
      'textures': textures
    };
  }

  static List<PlayerProfileTexturePropertyValue> listFromJson(List<dynamic> json) {
    return json == null ? new List<PlayerProfileTexturePropertyValue>() : json.map((value) => new PlayerProfileTexturePropertyValue.fromJson(value)).toList();
  }

  static Map<String, PlayerProfileTexturePropertyValue> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, PlayerProfileTexturePropertyValue>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new PlayerProfileTexturePropertyValue.fromJson(value));
    }
    return map;
  }
}

File: generated-sources/csharp-dotnet2/mojang-sessions/src/main/CsharpDotNet2/Org/OpenAPITools/Model/PlayerProfileTexturePropertyValue.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace Org.OpenAPITools.Model {

  /// <summary>
  /// This is obtained by decoding the \&quot;textures\&quot; property
  /// </summary>
  [DataContract]
  public class PlayerProfileTexturePropertyValue {
    /// <summary>
    /// UNIX timestamp in milliseconds
    /// </summary>
    /// <value>UNIX timestamp in milliseconds</value>
    [DataMember(Name="timestamp", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "timestamp")]
    public long? Timestamp { get; set; }

    /// <summary>
    /// The player UUID without hyphens
    /// </summary>
    /// <value>The player UUID without hyphens</value>
    [DataMember(Name="profileId", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "profileId")]
    public string ProfileId { get; set; }

    /// <summary>
    /// The player's name
    /// </summary>
    /// <value>The player's name</value>
    [DataMember(Name="profileName", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "profileName")]
    public string ProfileName { get; set; }

    /// <summary>
    /// Only present if unsigned was set to false in the request
    /// </summary>
    /// <value>Only present if unsigned was set to false in the request</value>
    [DataMember(Name="signatureRequired", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "signatureRequired")]
    public bool? SignatureRequired { get; set; }

    /// <summary>
    /// Gets or Sets Textures
    /// </summary>
    [DataMember(Name="textures", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "textures")]
    public PlayerTexture Textures { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class PlayerProfileTexturePropertyValue {\n");
      sb.Append("  Timestamp: ").Append(Timestamp).Append("\n");
      sb.Append("  ProfileId: ").Append(ProfileId).Append("\n");
      sb.Append("  ProfileName: ").Append(ProfileName).Append("\n");
      sb.Append("  SignatureRequired: ").Append(SignatureRequired).Append("\n");
      sb.Append("  Textures: ").Append(Textures).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
